src/modules/paymentProcessor/PP_Queue_v1.sol

```diff
@@ -0,0 +1,941 @@
+        _ensureValidClient(client_);
+        _;
+    }
+
+    // -------------------------------------------------------------------------
+    // Initialize
+
+    /// @inheritdoc Module_v1
+    function init(
+        IOrchestrator_v1 orchestrator_,
+        Metadata memory metadata_,
+        bytes memory configData_
+    ) external override(Module_v1) initializer {
+        __Module_init(orchestrator_, metadata_);
+
+        (uint maxQueueSize_, uint maxOrderLifetime_) =
+            abi.decode(configData_, (uint, uint));
+
+        if (maxQueueSize_ == 0) revert Module__PP_Queue_InvalidConfig();
+        if (maxOrderLifetime_ == 0) revert Module__PP_Queue_InvalidConfig();
+
+        _maxQueueSize = maxQueueSize_;
+        _maxOrderLifetime = maxOrderLifetime_;
+    }
+
+    //--------------------------------------------------------------------------
+    // Public (Getters)
+
+    /// @inheritdoc IPP_Queue_v1
+    function getOrder(uint orderId_)
+        external
+        view
+        returns (QueuedOrder memory order_)
+    {
+        if (!_orderExists(orderId_)) {
+            revert Module__PP_Queue_InvalidOrderId(
+                _orders[orderId_].client, orderId_
+            );
+        }
+        order_ = _orders[orderId_];
+    }
+
+    /// @inheritdoc IPP_Queue_v1
+    function getOrderQueue(address client_)
+        external
+        view
+        returns (uint[] memory queue_)
+    {
+        uint[] memory queue = new uint[](_queue[client_].length());
+        uint index_;
+
+        for (
+            uint id = _queue[client_].getNextId(LinkedIdList._SENTINEL);
+            id != LinkedIdList._SENTINEL;
+            id = _queue[client_].getNextId(id)
+        ) {
+            queue[index_++] = id;
+        }
+
+        queue_ = queue;
+    }
+
+    /// @inheritdoc IPP_Queue_v1
+    function getQueueHead(address client_) external view returns (uint head_) {
+        head_ = _queue[client_].getNextId(LinkedIdList._SENTINEL);
+    }
+
+    /// @inheritdoc IPP_Queue_v1
+    function getQueueTail(address client_) external view returns (uint tail_) {
+        tail_ = _queue[client_].lastId();
+    }
+
+    /// @inheritdoc IPP_Queue_v1
+    function getQueueSize(address client_)
+        external
+        view
+        returns (uint size_, uint maxSize_)
+    {
+        size_ = _queue[client_].length();
+        maxSize_ = _maxQueueSize;
+    }
+
+    /// @inheritdoc IPP_Queue_v1
+    function getTotalOrders() external view returns (uint total_) {
+        total_ = _nextOrderId;
+    }
+
+    /// @notice Gets the maximum size of the queue.
+    /// @return maxSize_ Maximum queue size.
+    function getMaxQueueSize() external view returns (uint maxSize_) {
+        maxSize_ = _maxQueueSize;
+    }
+
+    /// @notice Gets the maximum lifetime of an order.
+    /// @return maxLifetime_ Maximum order lifetime in seconds.
+    function getMaxOrderLifetime() external view returns (uint maxLifetime_) {
```

Comment:
Isn't used anywhere. What were the thoughts behind this ser

Reply:
I can't remember now, I've removed it for now